### ISSUE TITLE
[cgo] unloadLibrary is never called

### ISSUE THREAD

**Issue by alexanderoster** [Autodesk/AutomaticComponentToolkit#133]
We would need to call unload library with the wrapper destructor, but I am a bit helpless how this works. @qmuntal: Is there a chance you can take a look at this?

**Comment by qmuntal** [comment]
I don´t think it make sense to add the auto-release capability to the wrapper. It clearly a resource loaded by the user, therefore it should be unloaded by the user, same as when opening/closing files or http requests.

**Comment by alexanderoster** [comment]
I see, thank you for the feedback!
But does that mean the unload will leak memory if the wrapper goes out of scope?

**Comment by qmuntal** [comment]
> But does that mean the unload will leak memory if the wrapper goes out of scope?

Yes, unless the user calls `unload` before the wrapper goes out of scope. `defer` is very helpful in this scenarios:

```go
lib := act.LoadLibrary("C:/foo.dll")
defer lib.Unload() // lib will be automatically released when going out of scope
```
